examples/contrib/oed/gp_bayes_opt.py

```diff
@@ -0,0 +1,164 @@
+
+        :param str method: the method to use for acquisition. Choose from:
+            Thompson
+        :param int num_acquisitions: the number of points to generate
+        :param dict opt_params: additional parameters for optimization
+            routines
+        :return: a tensor of points to evaluate `self.f` at
+        :rtype: torch.Tensor
+        """
+
+        if method == "Thompson":
+
+            # Initialize the return tensor, add the UCB1 point
+            X = torch.zeros(num_acquisitions, *self.gpmodel.X.shape[1:])
+
+            for i in range(num_acquisitions):
+                sampler = self.gpmodel.iter_sample(noiseless=False)
+                x, _ = self.opt_differentiable(sampler, **opt_params)
+                X[i, ...] = x
+
+            return X
+
+        else:
+            raise NotImplementedError("Only method Thompson implemented for acquisition")
+
+    def run(self, num_steps, num_acquisitions):
+        """
+        Optimizes `self.f` in `num_steps` steps, acquiring `num_acquisitions`
+        new function evaluations at each step.
+
+        :param int num_steps:
+        :param int num_steps"
+        :return: the minimiser and the minimum value
+        :rtype: tuple
+        """
+
+        for i in range(num_steps):
+            X = self.acquire(num_acquisitions=num_acquisitions)
+            y = self.f(X)
+            self.update_posterior(X, y)
+
+        return self.opt_differentiable(lambda x: self.gpmodel(x)[0])
+
+    # def run(self, num_steps, num_acquisitions):
+    #     plt.figure(figsize=(12, 30))
+    #     outer_gs = gridspec.GridSpec(num_steps, 1)
+
+    #     for i in range(num_steps):
+    #         X = self.acquire(num_acquisitions=num_acquisitions)
+    #         y = self.f(X)
+    #         gs = gridspec.GridSpecFromSubplotSpec(1, 1, subplot_spec=outer_gs[i])
+    #         self.plot(gs, xlabel=i+1, with_title=(i % 2 == 0))
+    #         self.update_posterior(self.gpmodel, X, y)
+
+    #     plt.show()
+
+    #     return self.opt_differentiable(lambda x: self.gpmodel(x)[0])
+
+    # def plot(self, gs, xlabel=None, with_title=True):
+    #     xlabel = "xmin" if xlabel is None else "x{}".format(xlabel)
+    #     Xnew = torch.linspace(-1., 101.)
+    #     ax1 = plt.subplot(gs[0])
+    #     ax1.plot(self.gpmodel.X.detach().numpy(), self.gpmodel.y.detach().numpy(), "kx")  # plot all observed data
+    #     with torch.no_grad():
+    #         loc, var = self.gpmodel(Xnew, full_cov=False, noiseless=False)
+    #         sd = var.sqrt()
+    #         ax1.plot(Xnew.numpy(), loc.numpy(), "r", lw=2)  # plot predictive mean
+    #         ax1.fill_between(Xnew.numpy(), loc.numpy() - 2*sd.numpy(), loc.numpy() + 2*sd.numpy(),
+    #                          color="C0", alpha=0.3)  # plot uncertainty intervals
+    #     ax1.set_xlim(-1, 101)
+    #     ax1.set_title("Find {}".format(xlabel))
+    #     if with_title:
+    #         ax1.set_ylabel("Gaussian Process Regression")
```

Comment:
Commented code is unsightly but useful for debugging. `test_examples` will not admit `matplotlib` in example code.

Reply:
Feel free to use local imports for plotting dependencies:
```py
def plot(self, gs, ...):
    from matplotlib import pyplot as plt
    ...
```
That's not perfect, but it's better than commented-out code :wink:

Reply:
You should probably just add a separate test for this and remove the test code here

Reply:
I'll remove the code, keep it on a local branch